frame/system/src/lib.rs

```diff
@@ -513,6 +535,17 @@ decl_module! {
 			ensure_root(origin)?;
 			storage::unhashed::kill_prefix(&prefix);
 		}
+
+		/// Kill the sending account, assuming there are no references outstanding and the composite
+		/// data is equal to its default value.
+		#[weight = SimpleDispatchInfo::FixedOperational(25_000)]
+		fn suicide(origin) {
```

Comment:
```suggestion
		fn terminate(origin) {
```

I think we should learn from the Ethereum world and avoid the use of this word :/

The Contracts folks will also be using the term terminate as self-destruct for smart contracts.